frontend/src/Components/CardSelector.tsx

```diff
@@ -1,39 +1,47 @@
 import classNames from 'classnames';
-import { WebSocketApi } from '../types/WebSocket';
+import { CardValue, WebSocketApi } from '../types/WebSocket';
 import classes from './CardSelector.module.css';
 import { IconCoffee } from './IconCoffee';
 import { connectToWebSocket } from './WebSocket';
 import { IconObserver } from './IconObserver';
-import { BUTTON_OBSERVER, VOTE_COFFEE, VOTE_OBSERVER } from '../constants';
+import { BUTTON_OBSERVER, VOTE_COFFEE, VOTE_NOTE_VOTED, VOTE_OBSERVER } from '../constants';
+import { JSX } from 'preact';
+
+const SPECIAL_ICONS: { [value in CardValue]?: JSX.Element } = {
+  [VOTE_OBSERVER]: <IconObserver />,
+  [VOTE_COFFEE]: <IconCoffee />,
+};
+const getCard = (
+  cardValue: CardValue,
+  isSelected: boolean,
+  isObserver: boolean,
+  socket: WebSocketApi
```

Comment:
i would pass only setVote to the card so it must not know that it is used in a websocket context